bundles/org.openhab.core.io.rest/src/main/java/org/openhab/core/io/rest/Stream2JSONInputStream.java

```diff
@@ -32,74 +36,71 @@
  * nested collections JSON representation will be fully transformed into memory.
  *
  * @author Henning Treu - Initial contribution
+ * @author Jörg Sautter - Use as SequenceInputStream to simplify the logic
  */
 @NonNullByDefault
 public class Stream2JSONInputStream extends InputStream implements JSONInputStream {
 
-    private final Iterator<String> iterator;
-
-    private InputStream jsonElementStream;
-
-    private boolean firstIteratorElement;
-
-    private final Gson gson = new GsonBuilder().setDateFormat(DateTimeType.DATE_PATTERN_WITH_TZ_AND_MS).create();
+    private final InputStream stream;
 
     /**
      * Creates a new {@link Stream2JSONInputStream} backed by the given {@link Stream} source.
      *
      * @param source the {@link Stream} backing this input stream. Must not be null.
      */
     public Stream2JSONInputStream(Stream<?> source) {
-        iterator = source.map(e -> gson.toJson(e)).iterator();
-        jsonElementStream = new ByteArrayInputStream(new byte[0]);
-        firstIteratorElement = true;
-    }
+        Gson gson = new GsonBuilder().setDateFormat(DateTimeType.DATE_PATTERN_WITH_TZ_AND_MS).create();
```

Comment:
Why do we need a new instance on every call? Can't we share it as class field?

Reply:
Yes, that is correct.